chatkit-core/src/test/kotlin/com/pusher/chatkit/users/UserSubscriptionDispatcherTest.kt

```diff
@@ -0,0 +1,159 @@
+package com.pusher.chatkit.users
+
+import com.pusher.chatkit.rooms.api.JoinedRoomApiType
+import com.pusher.chatkit.rooms.api.JoinedRoomApiTypeMapper
+import com.pusher.chatkit.rooms.api.RoomMembershipApiType
+import com.pusher.chatkit.rooms.api.RoomReadStateApiType
+import com.pusher.chatkit.rooms.state.JoinedRoomsStateDiffer
+import com.pusher.chatkit.state.JoinedRoom
+import com.pusher.chatkit.state.JoinedRoomsReceived
+import com.pusher.chatkit.state.LeftRoom
+import com.pusher.chatkit.state.RoomDeleted
+import com.pusher.chatkit.state.RoomUpdated
+import com.pusher.chatkit.users.api.UserApiType
+import com.pusher.chatkit.users.api.UserSubscriptionDispatcher
+import com.pusher.chatkit.users.api.UserSubscriptionEvent
+import com.pusher.chatkit.util.DateApiTypeMapper
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.reduxkotlin.Dispatcher
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+class UserSubscriptionDispatcherTest : Spek({
```

Comment:
For consistency with existing tests (e.g. `UserSubscriptionEventParserSpec`) let's use `object`, not `class`. Conceptually it's also cleaner as it clearly conveys information that one instance is created during test execution.

Reply:
👍 sounds fair - have updated